lib/vdc/GridHelper.cpp

```diff
@@ -606,6 +606,11 @@ void GridHelper::_makeGridHelper(const DC::DataVar &var, const DimsType &roi_dim
         mv = var.GetMissingValue();
     }
 
+    // Don't allow NaNs as missing values. We replace any
+    // missing values in the data with infinity
+    //
+    if (std::isnan(mv)) mv = std::numeric_limits<float>::infinity();
```

Comment:
I assume there won't be incoming data sets with double type and also containing missing values? 

Also, NetCDFCFCollection::_GetMissingValue() seems to represent mv as a double. So maybe there needs some consistency improvement, or better, complete coverage improvement (i.e., properly handle both float and double types)?

Reply:
> I assume there won't be incoming data sets with double type and also containing missing values?

That is correct. The Grid class is not templated; data values are always single precision floats.

> 
> Also, NetCDFCFCollection::_GetMissingValue() seems to represent mv as a double. So maybe there needs some consistency improvement, or better, complete coverage improvement (i.e., properly handle both float and double types)?

Agreed. But I'm not sure that it is in scope for this fix.